File: mindelay/toolbox/IoU.py
import numpy as np


def IoU(box1, box2):
    I = max(min(box1[2], box2[2]) - max(box1[0], box2[0]), 0) * max(min(box1[3], box2[3]) - max(box1[1], box2[1]), 0)
    U = (box1[2] - box1[0]) * (box1[3] - box1[1]) + (box2[2] - box2[0]) * (box2[3] - box2[1]) - I
    return I / U

File: Evaluation_toolbox/IoU.m
function score = IoU(box1, box2)
% bounding boxes: [x1 y1 x2 y2]
x = min(box1(3),box2(3))-max(box1(1), box2(1));
y = min(box1(4),box2(4))-max(box1(2), box2(2));
if x > 0 && y > 0
    I = x*y;
    U = (box1(3)-box1(1))*(box1(4)-box1(2)) + (box2(3)-box2(1))*(box2(4)-box2(2)) - I;
    score = I/U;
else
    score = 0;
end
end
